### PR TITLE
Modified README.md file for OpenEBS pools & volumes monitoring

### PR DESCRIPTION
**Pull request by Ab-hishek**
Signed-off-by: Abhishek Agarwal <abhishek.agarwal@mayadata.io>

#### Special notes for your reviewer:
Modified README.md file to include the steps required for monitoring OpenEBS pools & volumes for different Prometheus setups. Also removed unwanted links from storagepoolclaim dashboard.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [ ] Chart Version bumped
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[stable/openebs]`)

### CODE REVIEW

**Comment by survivant** [comment]
I'll take a look

File: grafana-charts/README.md [hunk 1]
```diff
@@ -2,19 +2,234 @@
...
+2. Run `kubectl get pods -n openebs -o wide` to verify and to get where Prometheus and Grafana instances are running.
 3. Run `kubectl get nodes -o wide` to get the node’s IP.
-4. Run `kubectl get svc` and note down the port allocated to Prometheus and Grafana services.
+4. Run `kubectl get svc -n openebs` and note down the port allocated to Prometheus and Grafana services.
 5. Open your browser and open `<nodeip:nodeport>`(32514 for Prometheus and 32515 for Grafana) and you should be able to see the Prometheus’s expression browser and Grafana’s UI on your browser.
-6. Login into it(default username and password is admin), add data source name as prometheus, select datasource type Prometheus and pass the `<nodeip:nodeport>`of Prometheus in the url field and click on run and test.
+6. Login into it(default username and password is admin), add data source name as DS_OPENEBS_PROMETHEUS, select datasource type Prometheus and pass the `<nodeip:nodeport>`of Prometheus in the url field and click on run and test.
 7. Now go to the next tab (Dashboard) and import the desired dashboard.You should be able to get preloaded dashboard of prometheus.
```
Comment (survivant) [inline grafana-charts/README.md:15]: click on +, import

image

File: grafana-charts/README.md [hunk 2]
```diff
@@ -2,19 +2,234 @@
...
 8. To create openebs dashboard, go to the import and paste the json from this folder.
```
Comment (survivant) [inline grafana-charts/README.md:16]: To import OpenEBS dashboards, you will to import them separatly (please..fix my english. and typos.) You can upload the json files one by one, or open a json file and copy paste the content.   Also..specify where are the dashaobard files.. like 
clustervolumesdashboard.json  

image

select the datasource click import

image

repeat the process for all dashboard files

File: grafana-charts/README.md [hunk 3]
```diff
@@ -2,19 +2,234 @@
...
 9. You should be able to get the graph of openebs volumes in the UI.
 
+* When using default Prometheus setup(or your own setup):
+It can be done in 2 ways. First way is to follow the 9 steps given below or the second way is the add the custom rules for `openebs-volumes`, `openebs-pools` and `pv-exporter` to let Prometheus scrape metrics from them. For the later way you can refer to the `openebs-volumes`, `openebs-pools` and `pv-exporter` jobs described inside the `openebs-monitoring-pg.yaml`.
+
+1. Make sure that the cStor target pods or any other volumes(jiva or localpv) created by OpenEBS storage engines have the following annotations added to them:
+```
+prometheus.io/scrape: 'true'
+prometheus.io/path: '/data/metrics'
+prometheus.io/port: '80'
```
Comment (survivant) [inline grafana-charts/README.md:26]: just add a example how to get those annotations like a describe.

**Review by survivant — COMMENTED** [review]
I'll continue later

File: grafana-charts/README.md [hunk 4]
```diff
@@ -2,19 +2,234 @@
...
+- source_labels: [__address__, __meta_kubernetes_pod_annotation_prometheus_io_port]
+    action: replace
+    regex: ([^:]+)(?::\d+)?;(\d+)
+    replacement: $1:$2
+    target_label: __address__
+```
+Here, `__metrics_path__` and `port` and whether to scrap metrics from this pod are being read from pod annotations.
+
+2. Run `kubectl get nodes -o wide` to get the node’s IP.
+3. Run `kubectl get svc -n <namespace>` and note down the port allocated to Prometheus and Grafana services. Here `<namespace>` must be replaced with the namespace in which the Prometheus and Grafana services are running. If the services are not already created then create a new NodePOrt service for both of them.
```
Comment (survivant) [inline grafana-charts/README.md:81]: typo : NodePort

**Comment by survivant** [comment]
@kmova review done.

**Comment by Ab-hishek** [comment]
Thanks @survivant . Will address your review comments asap.

**Comment by Ab-hishek** [comment]
Addressed all the review comments.

**Review by survivant — APPROVED** [review]

**Pull request merged**